Formula/arangodb.rb

```diff
@@ -11,81 +12,98 @@ class Arangodb < Formula
-        system "make", "install"
-      end
+    (libexec/"arangodb2/bin").install resource("upgrade")
+
+    resource("arangodb2").stage do
+      ENV.libcxx
+      ENV.cxx11
 
-    else
-      # clang on 10.8 will still try to build against libstdc++,
-      # which fails because it doesn't have the C++0x features
-      # arangodb requires.
       args = %W[
         --disable-dependency-tracking
-        --prefix=#{prefix}
+        --prefix=#{libexec}/arangodb2
         --disable-relative
-        --datadir=#{share}
         --localstatedir=#{var}
+        --program-suffix=-2.8
       ]
 
-      args << "--program-suffix=-unstable" if build.head?
-
-      if ENV.compiler != :clang
-        ENV.append "LDFLAGS", "-static-libgcc -static-libstdc++"
+      if ENV.compiler == "gcc-6"
+        ENV.append "CXXFLAGS", "-O2 -g -fno-delete-null-pointer-checks"
+        inreplace "3rdParty/Makefile.v8", "CXXFLAGS=\"", "CXXFLAGS=\"-fno-delete-null-pointer-checks "
       end
 
       system "./configure", *args
       system "make", "install"
     end
-  end
 
-  # moving the "if" inside post_install does not work
-  if build.head?
-    def post_install
-      (var/"lib/arangodb3").mkpath
-      (var/"log/arangodb3").mkpath
-    end
-  else
-    def post_install
-      (var/"arangodb").mkpath
-      (var/"log/arangodb").mkpath
+    mkdir "build" do
+      args = std_cmake_args + %W[
+        -DHOMEBREW=ON
+        -DUSE_OPTIMIZE_FOR_ARCHITECTURE=OFF
+        -DASM_OPTIMIZATIONS=OFF
+        -DETCDIR=#{etc}
+        -DVARDIR=#{var}
+      ]
+
+      if ENV.compiler == "gcc-6"
+        ENV.append "V8_CXXFLAGS", "-O3 -g -fno-delete-null-pointer-checks"
+      end
 
-      system "#{sbin}/arangod", "--upgrade"
+      system "cmake", "..", *args
+      system "make", "V=1", "Verbose=1", "VERBOSE=1", "install"
     end
   end
 
+  def post_install
+    (var/"lib/arangodb3").mkpath
+    (var/"log/arangodb3").mkpath
+
+    args = %W[
+      #{libexec}/arangodb2
+      #{var}/lib/arangodb
+      #{HOMEBREW_PREFIX}/opt/arangodb
+      #{var}/lib/arangodb3
+    ]
+
+    system libexec/"arangodb2/bin/upgrade.sh", *args
+  end
+
   def caveats
     s = <<-EOS.undent
-      Please note that clang and/or its standard library 7.0.0 has a severe
-      performance issue. Please consider using '--cc=gcc-5' when installing
-      if you are running on such a system.
-    EOS
+      The database format between ArangoDB 2.x and ArangoDB 3.x has
+      been changed, please checkout
+      https://docs.arangodb.com/3.0/Manual/Administration/Upgrading/index.html
 
-    if build.head?
-      s += <<-EOS.undent
-        A default password has been set. You can change it by executing
-          #{sbin}/arango-secure-installation
-      EOS
-    end
+      An empty password has been set. Please change it by executing
+        #{sbin}/arango-secure-installation
```

Comment:
No change in functionality here but `opt_sbin` is a bit nicer to look at/memorise in printed caveats.